### PR TITLE
feat: support dataframe.cov

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Document:
https://screenshot.googleplex.com/9egi7MsNj2uWHkH

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by chelsea-lin** [comment]
- Do we want to support for `series.cov`? https://pandas.pydata.org/docs/reference/api/pandas.Series.cov.html
- Please also check if multi-index works as expected?
https://github.com/googleapis/python-bigquery-dataframes/pull/379/files#diff-5867532a2d403a1dffcfc627f18835fd75e24e87dfd2cdc56a6df86510a6ee65

File: bigframes/core/blocks.py [hunk 1]
```diff
@@ -1106,13 +1106,21 @@ def summarize(
...
+    def calculate_pairwise_metric(self, op=agg_ops.CorrOp()):
+        """
+        Returns a block object to compute pairwise metrics among all value columns in this block.
+
+        The metric to be computed is specified by the `op` parameter, which can be either a
+        correlation operation (default) or a covariance operation.
+        """
+        if len(self.value_columns) > 30:
+            raise NotImplementedError(
+                f"Only work with dataframes containing fewer than 30 columns. Current: {len(self.columns)}. {constants.FEEDBACK_LINK}"
```
Comment (chelsea-lin) [inline bigframes/core/blocks.py:1118]: please help to double check if error message matches with the code? Does "fewer than 30 columns" include 30?

Comment (Genesis929) [reply bigframes/core/blocks.py]: Updated

File: bigframes/dataframe.py [hunk 2]
```diff
@@ -1019,17 +1019,21 @@ def corr(self, method="pearson", min_periods=None, numeric_only=False) -> DataFr
...
             frame = self._raise_on_non_numeric("corr")
         else:
             frame = self._drop_non_numeric()
 
-        return DataFrame(frame._block.corr())
+        return DataFrame(frame._block.calculate_pairwise_metric(op=agg_ops.CorrOp()))
+
+    def cov(self, *, numeric_only: bool = False) -> DataFrame:
+        if not numeric_only:
+            frame = self._raise_on_non_numeric("corr")
```
Comment (chelsea-lin) [inline bigframes/dataframe.py:1032]: "cov"

**Review by chelsea-lin — APPROVED** [review]
Minor comments. Overall LGTM

**Pull request merged**